Improve self-hosted docs (#2553)
gitpod-io/gitpod: docs/self-hosted/self-hosted.md

## Installation

> You can find all the installation methods and relevant scripts under the [`install`](https://github.com/gitpod-io/gitpod/tree/master/install) directory.

Reviewer: This is the beginning of the funnel and here I thing we should not guide the users away from the docs (and into the source code). People should read the docs first and after they did it we can tell them "for further reading, see here is the source code and more READMEs".

> **Recommended installation**: We strongly recommend choosing GCP or AWS to deploy and run a Gitpod instance. You can use our Terraform scripts to configure, deploy, and upgrade an instance.

Reviewer 2: It might be worth mentioning the installers here, too ... they use Terraform under the hood, but make things much easier for those not familiar with that kind of tech.

> The state of deploying Gitpod on vanilla Kubernetes is not mature yet but you can try it by using the helm charts from the [`chart`](https://github.com/gitpod-io/gitpod/tree/master/chart) directory.

Author: **thought**: Got this from https://community.gitpod.io/t/multiple-installation-issues-0-5-0-vanilla-k3s-kubernetes/2221/2. Feel free to remove if you think this is redundant.

Reviewer 3: I like the *recommended* part. But allthough the current documentation is lacking we still hope to have sth better before :christmas_tree: , and want to advertise "vanilla Kubernetes" as valid install option.

Author: Thanks for taking a look @geropl! WDYT of rephrasing the second part to something like the following?

```suggestion
Alternatively, you can use the helm charts from the [`chart`](https://github.com/gitpod-io/gitpod/tree/master/chart) directory to configure and deploy Gitpod on Kubernetes.
```

Reviewer: For the 0.6.0 release the helm charts will be more mature so we don't need to recommend avoiding it.
